JPATask/src/main/java/com/instinctools/egor/mentoring/JPA/DAO/DAOImpl/ClientDAOImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.instinctools.egor.mentoring.JPA.DAO.DAOImpl;
+
+import com.instinctools.egor.mentoring.JPA.DAO.ClientDAO;
+import com.instinctools.egor.mentoring.JPA.DAO.DAOUtil.DAOUtil;
+import com.instinctools.egor.mentoring.JPA.Entity.Account;
+import com.instinctools.egor.mentoring.JPA.Entity.Client;
+
+import javax.persistence.EntityManager;
+import java.sql.SQLException;
+import java.util.List;
+
+public class ClientDAOImpl implements ClientDAO {
+    public void createClient(Client client) throws SQLException {
+        EntityManager session = DAOUtil.getSessionFactory().createEntityManager();
+        try {
+            session.getTransaction().begin();
+            session.persist(client);
+            session.getTransaction().commit();
+        }
+        finally {
+            if(session != null && session.isOpen()){
+                session.close();
+            }
+        }
+    }
+
+    public void deleteClient(Client client) throws SQLException {
+        EntityManager session = DAOUtil.getSessionFactory().createEntityManager();
+        try {
+            session.getTransaction().begin();
+            session.remove(session.contains(client) ? client : session.merge(client));
+            session.flush();
```

Comment:
As I remember - no need to do flush() - it will be done automatically during transaction commit